### PR TITLE
update core tools and .net version for v4 end2end testing

### PR DESCRIPTION
**Pull request by kaibocai**
update core tools to version 4 for end2end testing.

### CODE REVIEW

File: setup-tests.ps1 [hunk 1]
```diff
@@ -1,57 +0,0 @@
-
```
Comment (Francisco-Gamino) [inline setup-tests.ps1:1]: I was not aware this code is a dup of `setup-tests-pipeline.ps1`. Are we referencing `setup-tests.ps1` anywhere?

**Review by Francisco-Gamino — DISMISSED** [review]
One minor comment. Otherwise, LGTM.

File: setup-tests.ps1 (hunk 1 above)
Comment (kaibocai) [reply setup-tests.ps1:1]: It was referenced in build.ps1 which is not used anywhere in the project. I believe they are unused files (build.ps1 and setup-tests.ps1). @amamounelsayed please confirm it. Thank you.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -57,6 +57,17 @@ jobs:
           TargetFolder: '$(Build.ArtifactStagingDirectory)'
           CleanTargetFolder: true
         displayName: 'Copying files for artifacts'
+
+      - pwsh: |
+          Invoke-WebRequest 'https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.ps1' -OutFile 'dotnet-install.ps1'
+          ./dotnet-install.ps1 -InstallDir "$env:ProgramFiles/dotnet" -Version "6.0.100-preview.7.21379.14" -Channel 'release'
```
Comment (Francisco-Gamino) [inline azure-pipelines.yml]: Please update this version to be `6.0.100-rc.2.21505.57`. This is what the Core Tools is currently using https://github.com/Azure/azure-functions-core-tools/blob/v4.x/azure-pipelines.yml#L49.

Comment (kaibocai) [reply azure-pipelines.yml]: updated it.

Comment (Francisco-Gamino) [reply azure-pipelines.yml]: Great, thanks.

**Pull request closed without merging**